src/index.ts

```diff
@@ -42,26 +87,16 @@ export default class Common {
    * @returns The dictionary with parameters set as chain
    */
   setChain(chain: string | number | object): any {
-    if (typeof chain === 'number') {
-      if (chainParams['names'][chain]) {
-        this._chainParams = chainParams[chainParams['names'][chain]]
-      } else {
-        throw new Error(`Chain with ID ${chain} not supported`)
-      }
-    } else if (typeof chain === 'string') {
-      if (chainParams[chain]) {
-        this._chainParams = chainParams[chain]
-      } else {
-        throw new Error(`Chain with name ${chain} not supported`)
-      }
+    if (typeof chain === 'number' || typeof chain === 'string') {
+      this._chainParams = Common._getChainParams(chain)
     } else if (typeof chain === 'object') {
       const required = ['networkId', 'genesis', 'hardforks', 'bootstrapNodes']
```

Comment:
Can't you go along these `required` checks with the attribute definitions and make the others optional? Would this still have some breaking change character?

Reply:
The problem is that those are untyped. We could type the param with something like `{ networkId: any; genesis: any; hardforks: any; bootstrapNodes: any; }`, but it may be more confusing than helpful.

Reply:
Ah ok, then remove the type reference, will soon go to bed but can approve again if you can do the change within the next couple of minutes.

Reply:
I didn't include it. So will merge this now, and create the release PR.

Reply:
😄